### PR TITLE
Add support for Ubuntu Focal (20.04)

### PR DESCRIPTION
**Pull request by egordon**
Since we are pushing to migrate to Github Actions, we should fully add CI support for Focal.

See https://github.com/personalrobotics/aikido/pull/598

### CODE REVIEW

File: internal-convert.py [hunk 1]
```diff
@@ -3,10 +3,11 @@
 from urlparse import urlparse
 import sys
 import yaml
+from future.utils import iteritems
 
-distribution = yaml.load(sys.stdin)
+distribution = yaml.unsafe_load(sys.stdin)
```
Comment (brianhou) [inline internal-convert.py]: Why `yaml.unsafe_load(...)` vs `yaml.load(..., Loader=yaml.FullLoader)`?

File: internal-convert.py [hunk 2]
```diff
@@ -3,10 +3,11 @@
...
 
-for repository_name, repository in distribution['repositories'].iteritems():
+for repository_name, repository in iteritems(distribution['repositories']):
```
Comment (brianhou) [inline internal-convert.py]: Why `iteritems` vs just calling `.items`?

File: internal-distro.py [hunk 3]
```diff
@@ -88,22 +89,22 @@ def main():
...
-        for name, options in packages_raw.iteritems()
+        for name, options in iteritems(packages_raw)
     }
 
     # Build a map from package name to the repository that contains it, based
     # soley on the information in the distribution file.
     distribution_package_map = dict()
 
-    for repository in repositories.itervalues():
+    for repository in itervalues(repositories):
```
Comment (brianhou) [inline internal-distro.py]: `.values()`?

File: internal-distro.py [hunk 4]
```diff
@@ -202,7 +203,7 @@ def main():
                 print('    Found package "{:s}" => {:s}'.format(
                     installed_package.name, installed_package.location))
 
-            installed_packages.update(repository_package_map.iterkeys())
+            installed_packages.update(iterkeys(repository_package_map))
```
Comment (brianhou) [inline internal-distro.py]: `.keys()` if you want to be extra explicit, but just passing `repository_package_map` should work automatically.

**Review by brianhou — CHANGES_REQUESTED** [review]

File: internal-convert.py (hunk 2 above)
Comment (egordon) [reply internal-convert.py]: See https://www.python.org/dev/peps/pep-0469/

Specifically: "Migrating to the common subset of Python 2 and 3"

The code from future.utils implements the right if-then statements to ensure that this code works on both Python 2.6+ and Python 3.

File: internal-distro.py (hunk 3 above)
Comment (egordon) [reply internal-distro.py]: See prev comment

File: internal-distro.py (hunk 4 above)
Comment (egordon) [reply internal-distro.py]: iterkeys is redundant (again, as per PEP 469), but I see no reason to add more unnecessary syntatic sugar

File: internal-convert.py (hunk 2 above)
Comment (egordon) [reply internal-convert.py]: (This is necessary since we are still running Github Action checks on both Bionic [Python 2] and Focal [Python 3])

Comment (brianhou) [reply internal-convert.py]: Correct me if I'm wrong, but I don't think the iterator aspect of it is actually important in this refactor (unless it's being changed during iteration...), so just calling `.items` seems fine to me. That method is supported in both Python 2 and 3.

**Review by brianhou — APPROVED** [review]

**Pull request merged**